autogpt_platform/frontend/src/components/node-input-components.tsx

```diff
@@ -232,34 +248,131 @@ const NodeDateTimeInput: FC<{
+  error?: string;
+  handleInputChange: NodeObjectInputTreeProps["handleInputChange"];
+  className?: string;
+  displayName: string;
+}> = ({
+  selfKey,
+  schema,
+  value = "",
+  error,
+  handleInputChange,
+  className,
+  displayName,
+}) => {
+  const handleFileChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files?.[0];
+      if (!file) return;
+
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        const base64String = e.target?.result as string;
+        handleInputChange(selfKey, base64String);
+      };
+      reader.readAsDataURL(file);
+    },
+    [selfKey, handleInputChange],
+  );
+
+  const getFileLabel = useCallback((value: string) => {
+    if (value.startsWith("data:")) {
+      const matches = value.match(/^data:([^;]+);/);
+      if (matches?.[1]) {
+        const mimeParts = matches[1].split("/");
+        if (mimeParts.length > 1) {
+          return `${mimeParts[1].toUpperCase()} file`;
+        }
+        return `${matches[1]} file`;
+      }
+    } else {
+      const pathParts = value.split(".");
+      if (pathParts.length > 1) {
+        const ext = pathParts.pop();
+        if (ext) return `${ext.toUpperCase()} file`;
+      }
+    }
+    return "File";
+  }, []);
+
+  return (
+    <div className={cn("flex flex-col gap-2", className)}>
+      <div className="nodrag flex flex-col gap-2">
+        <div className="flex items-center gap-2">
           <Button
             variant="outline"
-            className={cn(
-              "w-full justify-start text-left font-normal",
-              !value && "text-muted-foreground",
-            )}
+            onClick={() =>
+              document.getElementById(`${selfKey}-upload`)?.click()
+            }
+            className="w-full"
           >
-            <CalendarIcon className="mr-2 h-4 w-4" />
-            {value ? format(date, "PPP") : <span>Pick a date</span>}
+            {value ? `Change ${displayName}` : `Upload ${displayName}`}
           </Button>
-        </PopoverTrigger>
-        <PopoverContent className="w-auto p-0" align="start">
-          <Calendar
-            mode="single"
-            selected={date}
-            onSelect={handleDateSelect}
-            initialFocus
-          />
-        </PopoverContent>
-      </Popover>
-      <LocalValuedInput
-        type="time"
-        value={timeInput}
-        onChange={handleTimeChange}
-        className="w-full"
-      />
+          {value && (
+            <Button
+              variant="ghost"
+              className="text-red-500 hover:text-red-700"
+              onClick={() => handleInputChange(selfKey, "")}
+            >
+              <Cross2Icon className="h-4 w-4" />
+            </Button>
+          )}
+        </div>
+        <input
+          id={`${selfKey}-upload`}
+          type="file"
+          accept="*/*"
+          onChange={handleFileChange}
+          className="hidden"
+        />
```

Comment:
https://ui.shadcn.com/docs/components/input#file

Reply:
The whole UI for the file input will be eventually redone, note that the actual input is hidden.